### PR TITLE
[AArch64][GlobalISel] Add support for extending indexed loads.

### PR DESCRIPTION
**Pull request by aemerson**
This is the final patch in the stack https://github.com/llvm/llvm-project/pull/69533

### CODE REVIEW

**Comment by aemerson** [comment]
The selection code is pretty much a straight port from `AArch64TargetLowering::SelectIndexedLoad()` except we split extload handling separately from regular indexed loads instead of one big function.

**Review by jroelofs — APPROVED** [review]
LGTM

File: llvm/lib/Target/AArch64/GISel/AArch64InstructionSelector.cpp [hunk 1]
```diff
@@ -5648,6 +5652,93 @@ MachineInstr *AArch64InstructionSelector::tryAdvSIMDModImmFP(
...
+    return false; // We should be on gpr.
+
+  auto Cst = getIConstantVRegVal(Offset, MRI);
+  if (!Cst)
+    return false; // Shouldn't happen, but just in case.
+
+  auto LdMI = MIB.buildInstr(Opc, {WriteBack, NewLdDstTy}, {Base})
+                  .addImm(Cst->getSExtValue());
+  LdMI.cloneMemRefs(ExtLd);
+  constrainSelectedInstRegOperands(*LdMI, TII, TRI, RBI);
```
Comment (tschuett) [inline llvm/lib/Target/AArch64/GISel/AArch64InstructionSelector.cpp:5722]: You ignore the `false`.

File: llvm/lib/Target/AArch64/GISel/AArch64InstructionSelector.cpp [hunk 2]
```diff
@@ -5648,6 +5652,93 @@ MachineInstr *AArch64InstructionSelector::tryAdvSIMDModImmFP(
...
+    // Generate a SUBREG_TO_REG.
+    auto SubToReg = MIB.buildInstr(TargetOpcode::SUBREG_TO_REG, {Dst}, {})
+                        .addImm(0)
+                        .addUse(LdMI.getReg(1))
+                        .addImm(AArch64::sub_32);
+    RBI.constrainGenericRegister(SubToReg.getReg(0), AArch64::GPR64RegClass,
+                                 MRI);
+  } else {
+    auto Copy = MIB.buildCopy(Dst, LdMI.getReg(1));
+    selectCopy(*Copy, TII, MRI, TRI, RBI);
```
Comment (tschuett) [inline llvm/lib/Target/AArch64/GISel/AArch64InstructionSelector.cpp:5735]: You ignore the `false`.

File: llvm/lib/Target/AArch64/GISel/AArch64InstructionSelector.cpp (hunk 1 above)
Comment (tschuett) [reply llvm/lib/Target/AArch64/GISel/AArch64InstructionSelector.cpp:5722]: At least `constrainSelectedInstRegOperands` should be `[[nodiscard]]`.

**Pull request merged**